src/common/AlertItem.js

```diff
@@ -0,0 +1,101 @@
+/* @flow strict-local */
+
+import * as React from 'react';
+import { View } from 'react-native';
+// $FlowFixMe[untyped-import]
+import Color from 'color';
+
+import type { LocalizableReactText } from '../types';
+import { ThemeContext } from '../styles/theme';
+import { kNiceGreenColor, kWarningColor } from '../styles/constants';
+import { createStyleSheet } from '../styles';
+import ZulipTextIntl from './ZulipTextIntl';
+import ZulipTextButton from './ZulipTextButton';
+
+type Button = {|
+  +id: string,
+  +label: LocalizableReactText,
+  +inProgress?: boolean,
+  +onPress: () => Promise<void> | void,
+|};
+
+type Props = $ReadOnly<{|
+  text: LocalizableReactText,
+  buttons?: $ReadOnlyArray<Button>,
+
+  type: 'good' | 'neutral' | 'warn',
+
+  /**
+   * Give a top margin of the correct in-between space for a set of buttons
+   */
+  topMargin?: true,
+
+  /**
+   * Give a bottom margin of the correct in-between space for a set of
+   *   buttons
+   */
+  bottomMargin?: true,
+|}>;
+
+// TODO before reuse: jsdoc
+export default function AlertItem(props: Props): React.Node {
```

Comment:
(Skipping this file for now, because I'm not sure if it will still appear in the next revision of the UI.)